caddyfile.go

```diff
@@ -30,34 +32,45 @@ import (
 // See the godoc on the LogEncoderConfig type for the syntax of
 // subdirectives that are common to most/all encoders.
 func (se *TransformEncoder) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
-	foundTemplate := 0
-outerloop:
 	for d.Next() {
 		args := d.RemainingArgs()
 		switch len(args) {
 		case 0:
 			se.Template = commonLogFormat
 		default:
-			foundTemplate = len(args)
 			se.Template = strings.Join(args, " ")
 		}
 
 		for nesting := d.Nesting(); d.NextBlock(nesting); {
-			if d.Val() == "placeholder" {
+			switch d.Val() {
+			case "placeholder":
 				d.AllArgs(&se.Placeholder)
 				// delete the `placeholder` token and the value, and reset the cursor
 				d.Delete()
 				d.Delete()
-				break outerloop
+			case "unescape_strings":
+				// require an argument
+				if !d.NextArg() {
+					return d.ArgErr()
+				}
+				b, err := strconv.ParseBool(d.Val())
```

Comment:
In Caddyfile format, it's not typical for boolean toggles to require an argument. Their absence implies `false`, so their presence automatically implies `true` without the need for boolean argument.

Reply:
Good point :)
Change done